Tasks/AzureVmssDeployment/operations/VirtualMachineScaleSet.ts

```diff
@@ -12,10 +12,11 @@ export default class VirtualMachineScaleSet {
         this.taskParameters = taskParameters;
     }
 
-    public execute(): Promise<void> {
+    public async execute(): Promise<void> {
         var client = new armCompute.ComputeManagementClient(this.taskParameters.credentials, this.taskParameters.subscriptionId);
-        return new Promise<void>((resolve, reject) => {
-            this._getResourceGroupForVmss(client,resolve, reject, (error, result, request, response) => {
+        return new Promise<void>(async (resolve, reject) => {
```

Comment:
you can remove creating and returning this promise as well since your method is async